### PR TITLE
sync web engine; run web engine tests

### PR DESCRIPTION
**Pull request by yjbanov**
This PR pulls the latest version of the Web engine and enables Web engine tests on Cirrur.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/compositor/fonts.dart [hunk 1]
```diff
@@ -0,0 +1,89 @@
+part of engine;
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/compositor/fonts.dart]: Missing license header

File: lib/web_ui/lib/src/engine/compositor/image.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
+part of engine;
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/compositor/image.dart]: Missing license header

File: lib/web_ui/lib/src/engine/compositor/util.dart [hunk 3]
```diff
@@ -0,0 +1,272 @@
+part of engine;
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/compositor/util.dart]: Missing license header

File: lib/web_ui/lib/src/engine/pointer_binding.dart [hunk 4]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2013 The Flutter Authors. All rights reserved.
+// Copyright 2018 The Flutter Authors. All rights reserved.
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/pointer_binding.dart]: Don't change license dates

File: lib/web_ui/lib/src/engine/shader.dart [hunk 5]
```diff
@@ -0,0 +1,207 @@
+part of engine;
```
Comment (jonahwilliams) [inline lib/web_ui/lib/src/engine/shader.dart]: Missing license header

File: lib/web_ui/test/alarm_clock_test.dart [hunk 6]
```diff
@@ -1,7 +1,3 @@
-// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (jonahwilliams) [inline lib/web_ui/test/alarm_clock_test.dart]: Re-add this license header

File: .cirrus.yml [hunk 7]
```diff
@@ -35,6 +35,10 @@ task:
       test_host_script: |
         cd $ENGINE_PATH/src
         ./flutter/testing/run_tests.sh host_debug_unopt
+      test_web_engine: |
```
Comment (dnfield) [inline .cirrus.yml]: this has to be postfixed `_script`, so `test_web_engine_script`.

File: lib/web_ui/lib/src/engine/compositor/fonts.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/compositor/fonts.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/image.dart (hunk 2 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/compositor/image.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/util.dart (hunk 3 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/compositor/util.dart]: Done.

File: lib/web_ui/lib/src/engine/pointer_binding.dart (hunk 4 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/pointer_binding.dart]: Done.

File: lib/web_ui/lib/src/engine/shader.dart (hunk 5 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/shader.dart]: Done.

File: lib/web_ui/test/alarm_clock_test.dart (hunk 6 above)
Comment (yjbanov) [reply lib/web_ui/test/alarm_clock_test.dart]: Done.

File: .cirrus.yml (hunk 7 above)
Comment (yjbanov) [reply .cirrus.yml]: Thanks! Done.

**Comment by yjbanov** [comment]
@jonahwilliams Also added a license test so this doesn't happen in the future.

**Comment by dnfield** [comment]
@yjbanov - I would recommend investing time in gettin gthese tests running on LUCI rather than on Cirrus at this point, if they're not already running on LUCI.

**Comment by yjbanov** [comment]
@dnfield How do I configure LUCI to run these tests?

**Comment by dnfield** [comment]
https://github.com/flutter/flutter/tree/master/dev/bots has some docs that might help, but they're probably partly out of date at this point. We can sync offline too on this.

File: tools/licenses/lib/main.dart [hunk 8]
```diff
@@ -2040,10 +2043,40 @@ class _RepositoryFlutterDirectory extends _RepositoryDirectory {
...
+
+  final fs.Directory libRoot;
+
+  @override
+  bool shouldRecurse(fs.IoNode entry) {
+    final String relativePath = path.relative(entry.fullName, from: libRoot.fullName);
+    if (_kBlacklist.contains(relativePath)) {
+      return false;
+    }
+    return entry.name != 'toolchain'
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply tools/licenses/lib/main.dart]: Done.

**Pull request merged**